applications/GeoMechanicsApplication/custom_elements/small_strain_U_Pw_diff_order_element.cpp

```diff
@@ -850,13 +850,11 @@ void SmallStrainUPwDiffOrderElement::CalculateOnIntegrationPoints(const Variable
 {
     KRATOS_TRY
 
-    const unsigned int& IntegrationPointsNumber =
+    const auto number_of_integration_points =
         GetGeometry().IntegrationPointsNumber(this->GetIntegrationMethod());
 
-    if (rValues.size() != IntegrationPointsNumber) {
-        rValues.resize(IntegrationPointsNumber);
-    }
-    for (unsigned int i = 0; i < IntegrationPointsNumber; ++i) {
+    rValues.resize(number_of_integration_points);
```

Comment:
Just out of curiosity, do you know if the resize checks if it's necessary to do it? (In any case I don't think it really improves performance if we manually check like we did before, but I don't know enough about boost::ublas)

Reply:
First of all, note that `rValues` has type `std::vector<int>&`, which means that we need to refer to the STL documentation. [There](https://en.cppreference.com/w/cpp/container/vector/resize) it is says:

> Resizes the container to contain `count` elements, does nothing if `count == size()`.

In other words, the caller doesn't need to explicitly check the container's size. Calling `resize` on a `std::vector` that already has the desired size, has no effect.

Reply:
Perfect, thanks!